pkg/controller/node/nodecontroller.go

```diff
@@ -555,6 +565,88 @@ func (nc *NodeController) Run(stopCh <-chan struct{}) {
 	<-stopCh
 }
 
+// addOrRemoveNoScheduleTaintByCondition taint or remove NoSchedule taint on this node based on it's condition change.
+// return:
+// 1. if node changed to schedulable
+// 2. if node changed to unschedulable
+func (nc *NodeController) addOrRemoveNoScheduleTaintByCondition(node *v1.Node) (bool, bool) {
```

Comment:
[update] will update this to check existing node status only, so return a single value will be fine